gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionCompletionContributor.java

```diff
@@ -104,52 +113,82 @@ private void addLoadingElement(CompletionResultSet sortedResultSet) {
-                .flatMap(url -> StreamEx.of(
-                                groupPartOrPackageNameExplorer.getCancelableGroupPartOrPackageName(group, url))
+                .flatMap(url -> StreamEx.of(groupPartOrPackageNameExplorer.getCancelableGroupPartOrPackageName(
+                                dependencyInfo.group(), url))
                         .filter(pkgName -> pkgName.name().startsWith(dependencyNamePrefix))
                         .map(pkgName -> new SimpleEntry<>(url, pkgName)))
                 .map(entry -> {
                     RepositoryUrl url = entry.getKey();
                     GroupPartOrPackageName pkgName = entry.getValue();
                     DependencyName depName = DependencyName.of(pkgName.name());
-                    return new PackageInRepo(group, depName, url);
+                    return new PackageInRepo(dependencyInfo.group(), depName, url);
                 })
                 .toList();
     }
 
-    private void addToResults(CompletionResultSet resultSet, List<PackageInRepo> groupAndDeps) {
-        Map<DependencyVersion, AtomicInteger> versionCounts = StreamEx.of(groupAndDeps)
-                .flatMap(groupAndDep ->
-                        versionExplorer
-                                .getVersions(groupAndDep, CompletionRefreshUtil.refreshOnceSupplier()::get)
-                                .stream())
-                .collect(Collectors.toConcurrentMap(
-                        Function.identity(), v -> new AtomicInteger(1), (existingCount, newCount) -> {
-                            existingCount.addAndGet(newCount.get());
-                            return existingCount;
-                        }));
+    private void addToResults(CompletionResultSet resultSet, List<PackageInRepo> packageInRepo, DependencyInfo key) {
+
+        Set<VersionResults> versionResults =
+                packageInRepo.stream().map(versionExplorer::getVersions).collect(Collectors.toSet());
+
+        List<CompletableFuture<?>> pendingFutures = versionResults.stream()
+                .filter(results -> results.versions().isEmpty())
+                .map(VersionResults::future)
+                .filter(future -> !future.isDone())
+                .collect(Collectors.toList());
+
+        if (!pendingFutures.isEmpty()) {
+            CompletableFuture.anyOf(pendingFutures.toArray(new CompletableFuture[0]))
+                    .thenAccept(completedFuture -> {
+                        CompletionRefreshUtil.triggerRefresh();
+                    });
+        }
+
+        Map<DependencyVersion, Integer> versionCounts = StreamEx.of(versionResults)
+                .flatMap(result -> result.versions().stream())
+                .collect(Collectors.toMap(Function.identity(), v -> 1, Integer::sum));
+
+        if (pendingFutures.isEmpty() && versionCounts.isEmpty()) {
+            addDisplayElement(resultSet, "No versions found");
+            addAndRefresh(key);
+        }
+
+        if (!versionCounts.isEmpty()) {
```

Comment:
removes `Loading Versions...` from list of versions which is not strictly nessary as it sinks to the bottom but can be nicer when only a few versions are suggested